homeassistant/components/modbus/sensor.py

```diff
@@ -156,6 +160,5 @@ async def async_added_to_hass(self) -> None:
     def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         result = self.coordinator.data
-        if result:
-            self._attr_native_value = result[self._idx]
+        self._attr_native_value = result[self._idx] if result else None
```

Comment:
I have a suspicion this should be done to slaves in other filles like binary sensor, but I am trying to keep this file targeted as I am not set up to test binary sensor.  I will leave that for another dev.

Reply:
you are actually changing the functionality, using none might be a problem for people who have added automations.

Reply:
It starts at None.
When data is received it gets set.
The parent correctly sets to None, but the slave incorrectly keeps the last value when the connection fails.
This is inconsistent and wrong.
See parent:
if self._coordinator:
            if result:
                result_array = list(map(float if self._precision else int, result.split(",")))
                self._attr_native_value = result_array[0]
                self._coordinator.async_set_updated_data(result_array)
            else:
                self._attr_native_value = None
                self._coordinator.async_set_updated_data(None)

Reply:
How to test:
Define a sensor but keep it powered down.
-->parent=Unavailable, slave=Unknown
Power up sensor
-->parent=0.04, slave=0.04
Power down sensor (previous behaviour)
-->parent=Unavailable, slave=0.04
Power down sensor (updated behaviour)
-->parent=Unavailable, slave=Unknown
So updated behaviour is now consistent between being off when system started and being turned off later

Reply:
Good argument.